birthday/birthday.py

```diff
@@ -0,0 +1,284 @@
+        self.config.init_custom(self.DATE_GROUP, 1)
+        self.config.init_custom(self.GUILD_DATE_GROUP, 2)
+        self.config.register_guild(channel=None, role=None, yesterdays=[])
+        self.bday_loop = asyncio.ensure_future(self.initialise())
+        asyncio.ensure_future(self.check_breaking_change())
+
+    # Events
+    async def initialise(self):
+        await self.bot.wait_until_ready()
+        with contextlib.suppress(RuntimeError):
+            while self == self.bot.get_cog(self.__class__.__name__):
+                now = datetime.datetime.utcnow()
+                tomorrow = (now + datetime.timedelta(days=1)).replace(hour=0, minute=0, second=0, microsecond=0)
+                await self.clean_yesterday_bdays()
+                await self.do_today_bdays()
+                await asyncio.sleep((tomorrow - now).total_seconds())
+
+    def cog_unload(self):
+        self.bday_loop.cancel()
+
+    # Commands
+    @commands.group()
+    @commands.guild_only()
+    async def bday(self, ctx: Context):
+        """Birthday settings"""
+        pass
+
+    @bday.command(name="channel")
+    @checks.mod_or_permissions(manage_roles=True)
+    async def bday_channel(self, ctx: Context, channel: discord.TextChannel):
+        """Sets the birthday announcement channel"""
+        message = ctx.message
+        guild = message.guild
+        await self.config.guild(channel.guild).channel.set(channel.id)
+        await message.channel.send(self.CHANNEL_SET(g=guild.name, c=channel.name))
+
+    @bday.command(name="role")
+    @checks.mod_or_permissions(manage_roles=True)
+    async def bday_role(self, ctx: Context, *, role: discord.Role):
+        """Sets the birthday role"""
+        message = ctx.message
+        guild = message.guild
+        await self.config.guild(role.guild).role.set(role.id)
+        await message.channel.send(self.ROLE_SET(g=guild.name, r=role.name))
+
+    @bday.command(name="remove", aliases=["del", "clear", "rm"])
+    async def bday_remove(self, ctx: Context):
+        """Unsets your birthday date"""
+        message = ctx.message
+        await self.remove_user_bday(message.guild.id, message.author.id)
+        await message.channel.send(self.BDAY_REMOVED())
+
+    @bday.command(name="set")
+    async def bday_set(self, ctx: Context, *, date: str):
+        """Sets your birthday date
+
+        The given date can either be month day, or day month
+        Year is optional. If not given, the age won't be displayed."""
+        message = ctx.message
+        channel = message.channel
+        author = message.author
+        year = None
+        birthday = self.parse_date(date)
+        # An Invalid date was entered.
+        if birthday is None:
+            print(self.BDAY_INVALID())
+            await channel.send(self.BDAY_INVALID())
+        else:
+            print(type(birthday))
+            if datetime.datetime.utcnow().year != birthday.year:
+                year = birthday.year
```

Comment:
I would check and make sure no one sets a date year in the future or 1000 years in the past. Keep a limit of lets say, 100 years old.